openeo/utils/http.py

```diff
@@ -0,0 +1,93 @@
+"""
+openEO-oriented HTTP utilities
+"""
+
+from typing import Collection, Union
+
+import requests
+import requests.adapters
+from urllib3.util import Retry
+
+DEFAULT_RETRIES_TOTAL = 5
+
+# On `backoff_factor`: it influences how much to sleep according to the formula:
+#     sleep = {backoff factor} * (2 ** ({consecutive errors - 1}))
+# The sleep before the first retry will be skipped however.
+# For example with backoff_factor=2.5, the sleeps between consecutive attempts would be:
+#     0, 5, 10, 20, 40, ...
+DEFAULT_BACKOFF_FACTOR = 2.5
+
+
+DEFAULT_RETRY_FORCELIST = frozenset(
+    [
+        429,  # Too Many Requests
+        500,  # Internal Server Error
```

Comment:
Maybe it's best to not include `Internal Server Error` by default for now.
Unlike the others (429/502/503/504)   this is a very general error situation, and not something where we can assume that just waiting a bit and retrying will resolve the problem.